src/cgame/default/cg_hud.c

```diff
@@ -1175,6 +1233,231 @@ static void Cg_DrawTargetName(const player_state_t *ps) {
 /**
  * @brief
  */
+void Cg_ParseNotification(void) {
+	bg_notification_item_t *item = g_malloc0(sizeof(bg_notification_item_t));
+
+	item->type = cgi.ReadByte();
+
+	switch(item->type) {
+		case NOTIFICATION_TYPE_OBITUARY:
+			item->mod = cgi.ReadByte();
+			item->client_id_1 = cgi.ReadByte();
+			item->client_id_2 = cgi.ReadByte();
+
+			item->when = 6000;
+
+			if (!cg_draw_notifications_disable_print->value) {
+				cgi.Print("^7%s ^1%s ^7%s\n", cgi.client->client_info[item->client_id_1].name,
+					Bg_GetModString(item->mod, item->mod & MOD_FRIENDLY_FIRE),
+					cgi.client->client_info[item->client_id_2].name);
+			}
+
+			break;
+		case NOTIFICATION_TYPE_OBITUARY_SELF:
+			item->mod = cgi.ReadByte();
+			item->client_id_1 = cgi.ReadByte();
+
+			item->when = 6000;
+
+			if (!cg_draw_notifications_disable_print->value) {
+				cgi.Print("^1%s ^7%s\n", Bg_GetModString(item->mod, false),
+					cgi.client->client_info[item->client_id_1].name);
+			}
+
+			break;
+		case NOTIFICATION_TYPE_OBITUARY_PIC: {
+			const char *p = cgi.ReadString();
+			strcpy(item->pic, p);
+			item->client_id_1 = cgi.ReadByte();
+			item->client_id_2 = cgi.ReadByte();
+
+			item->when = 6000;
+
+			if (!cg_draw_notifications_disable_print->value) {
+				cgi.Print("^7%s ^1[%s] ^7%s\n", cgi.client->client_info[item->client_id_1].name,
+					item->pic,
+					cgi.client->client_info[item->client_id_2].name);
+			}
+		}
+			break;
+		case NOTIFICATION_TYPE_PLAYER_ACTION: {
+			const char *p = cgi.ReadString();
+			strcpy(item->pic, p);
+			item->client_id_1 = cgi.ReadByte();
+			const char *s = cgi.ReadString();
+			strcpy(item->string_1, s);
+
+			item->when = 8000;
+
+			if (!cg_draw_notifications_disable_print->value) {
+				cgi.Print("^7%s ^7%s\n", cgi.client->client_info[item->client_id_1].name,
+					item->string_1);
+			}
+		}
+			break;
+		default:
+			cgi.Warn("Invalid notification type %d\n", item->type);
+			break;
+	}
+
+	item->when = cgi.client->unclamped_time + (item->when * cg_draw_notifications_time->value);
+
+	notifications.items = g_slist_prepend(notifications.items, item);
```

Comment:
Items are always added, they're just not rendered.
I should probably make them not add if there isn't room or you could get notifications popping back into the screen sometimes.